diff --git a/BUILD.gn b/BUILD.gn
index ced0f3c..fcc56d2 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -1401,6 +1401,13 @@ template("angle_libGLESv2") {
       output_extension = angle_glesv2_extension
     }
 
+    if (is_mac && !is_component_build) {
+      ldflags = [
+        "-install_name",
+        "@rpath/${output_name}.dylib"
+      ]
+    }
+
     configs += [
       ":angle_gl_visibility_config",
       ":debug_annotations_config",
@@ -1472,6 +1479,13 @@ angle_shared_library("libGLESv1_CM") {
   sources = libglesv1_cm_sources
   output_name = "libGLESv1_CM${angle_libs_suffix}"
 
+  if (is_mac && !is_component_build) {
+    ldflags = [
+      "-install_name",
+      "@rpath/${output_name}.dylib"
+    ]
+  }
+
   configs += [
     ":angle_gl_visibility_config",
     ":debug_annotations_config",
@@ -1535,6 +1549,13 @@ template("libEGL_shared_template") {
       output_extension = angle_egl_extension
     }
 
+    if (is_mac && !is_component_build) {
+      ldflags = [
+        "-install_name",
+        "@rpath/${output_name}.dylib"
+      ]
+    }
+
     defines = invoker.defines + [ "LIBEGL_IMPLEMENTATION" ]
     if (is_win) {
       defines += [ "EGLAPI=" ]
