diff --git a/BUILD.gn b/BUILD.gn
index 08bd22a..d851f95 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -51,6 +51,13 @@ if (!is_android) {
       configs -= [ "//build/config/compiler:chromium_code" ]
       configs += [ "//build/config/compiler:no_chromium_code" ]
     }
+
+    if (is_apple && !is_component_build) {
+      ldflags = [
+        "-install_name",
+        "@rpath/lib${target_name}.dylib",
+      ]
+    }
   }
 
   action("vulkan_gen_icd_json_file") {
